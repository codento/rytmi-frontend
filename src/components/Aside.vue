<template>
  <aside class="aside-menu" />
</template>

<script>
export default {
  name: 'c-aside'
}
</script>
