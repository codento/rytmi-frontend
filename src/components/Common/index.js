import ProjectRow from './ProjectRow.vue'
import SkillRow from './SkillRow.vue'
import SkillExplanations from './SkillExplanations.vue'
import UtilizationChart from './UtilizationChart.vue'
import CollapsableItem from './CollapsableItem.vue'
import EducationCard from './EducationCard.vue'
import ProjectForm from './ProjectForm.vue'
import ProjectProfileForm from './ProjectProfileForm.vue'

export {
  ProjectRow,
  SkillRow,
  SkillExplanations,
  UtilizationChart,
  CollapsableItem,
  EducationCard,
  ProjectForm,
  ProjectProfileForm
}
