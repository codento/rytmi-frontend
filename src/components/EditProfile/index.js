import EditLinks from './EditLinks.vue'
import EditSkills from './EditSkills.vue'
import ProfileForm from './ProfileForm.vue'
import EditProjects from './EditProjects.vue'
import WorkHistoryEntryEdit from './WorkHistoryEntryEdit.vue'
import WorkHistoryEntryList from './WorkHistoryEntryList.vue'
import CvInfo from './CvInfo.vue'
import CvInfoEditIntroduction from './CvInfoEditIntroduction.vue'
import CvInfoEditEducation from './CvInfoEditEducation.vue'
import CvInfoEditEducationForm from './CvInfoEditEducationForm.vue'

export {
  EditLinks,
  EditSkills,
  ProfileForm,
  EditProjects,
  WorkHistoryEntryList,
  CvInfo,
  CvInfoEditIntroduction,
  CvInfoEditEducation,
  CvInfoEditEducationForm,
  WorkHistoryEntryEdit
}
