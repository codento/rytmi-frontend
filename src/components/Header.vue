<template>
  <header class="app-header navbar">
    <button class="navbar-toggler mobile-sidebar-toggler d-lg-none" type="button" @click="mobileSidebarToggle">
      <span class="navbar-toggler-icon"></span>
    </button>
    <b-link class="navbar-brand" to="Home"></b-link>
    <button class="navbar-toggler sidebar-toggler d-md-down-none" type="button" @click="sidebarToggle">
      <span class="navbar-toggler-icon"></span>
    </button>
     <b-navbar-nav  v-if="isAuthenticated">
      <b-nav-item to="/search">Search</b-nav-item>

      <b-nav-item v-if="!getProfileId" to="/createProfile">Create Profile</b-nav-item>
      <b-nav-item v-else to="/editProfile">Edit My Profile</b-nav-item>

    </b-navbar-nav>
    <b-navbar-nav class="ml-auto">
      <b-nav-item id="loginBtn"  v-if="!isAuthenticated" v-on:click="login" >Sign in</b-nav-item>
      <b-nav-item id="logoutBtn" v-if="isAuthenticated" v-on:click="logout" >Sign out</b-nav-item>
      <b-nav-item id="logoutBtn"  v-on:click="checkStatus" >Check status</b-nav-item>
    </b-navbar-nav>
    <button class="navbar-toggler aside-menu-toggler d-md-down-none" type="button" @click="asideToggle">
      <span class="navbar-toggler-icon"></span>
    </button>
    
  </header>
</template>
<script>
import {mapGetters, mapActions} from 'vuex'
import {loadAuthClient, handleLogin, handleLogout} from '../utils/auth'

export default {
  name: 'c-header',
  computed: {
    ...mapGetters(['isAuthenticated', 'getProfileId', 'getToken'])
  },
  methods: {
    sidebarToggle (e) {
      e.preventDefault()
      document.body.classList.toggle('sidebar-hidden')
    },
    sidebarMinimize (e) {
      e.preventDefault()
      document.body.classList.toggle('sidebar-minimized')
    },
    mobileSidebarToggle (e) {
      e.preventDefault()
      document.body.classList.toggle('sidebar-mobile-show')
    },
    asideToggle (e) {
      e.preventDefault()
      document.body.classList.toggle('aside-menu-hidden')
    },
    checkStatus () {
      this.$toasted.global.rytmi_success({
        message: 'Authenticated: ' + this.isAuthenticated
      })
    },
    logout () {
      handleLogout().then(() => {
        this.logoutAuth()
        this.$router.push(this.$route.query.redirect || '/home')
      })
    },
    login () {
      handleLogin().then((response) => {
        this.requestAuth(response.Zi.id_token)
      }).catch((error) => {
        console.log(error)
        this.$toasted.global.rytmi_error({
          message: 'Login failed'
        })
      })
    },
    ...mapActions(['requestAuth', 'logoutAuth', 'createUser'])
  },
  created () {
    /* eslint-disable */
    loadAuthClient()
  }
}
</script>
