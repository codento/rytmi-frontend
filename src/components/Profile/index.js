import ProjectRow from './ProjectRow.vue'
import SkillExplanations from './SkillExplanations.vue'
import SkillRow from './SkillRow.vue'
import UserProfileCard from './UserProfileCard.vue'

export {
  ProjectRow,
  SkillExplanations,
  SkillRow,
  UserProfileCard
}
