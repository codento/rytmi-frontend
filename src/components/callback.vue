<template>
  
</template>

<script>
import {mapActions} from 'vuex'

export default {
  created () {
    this.$nextTick(() => {
      this.fetchProfiles()
      this.fetchSkills()
      this.fetchProfileSkills()
      this.$router.push(this.$route.query.redirect || '/')
    })
  },
  methods: {
    ...mapActions([
      'fetchProfiles',
      'fetchSkills',
      'fetchProfileSkills'
    ])
  }
}
</script>