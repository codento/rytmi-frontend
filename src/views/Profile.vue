<template>
  <div class="animated fadeIn">
    <b-row>
      <b-col>
        <UserProfileCard :basicInfo="basicInfo"/>
      </b-col>
      <b-col>
        <b-card header='Taidot <i class="float-right fa fa-edit"></i>' >
          <div class="row mb-1">
            <SkillRow class="col-sm-6"  v-for='skill in sortSkills' :name="skill.name" v-bind:item="skill" :proficiency="skill.knows" :key='skill.id' />
          </div>
        </b-card>
        <b-card header='Työkokemus'>
          <ProjectRow v-for='project in projects' :project="project" :projectName="project.projectName" :key="project.projectId"/>
        </b-card>
      </b-col>
    </b-row>
  </div>
</template>

<script>
import SkillRow from '../components/SkillRow'
import ProjectRow from '../components/ProjectRow'
import UserProfileCard from '../components/UserProfileCard'

export default {
  name: 'Profile',
  computed: {
    sortSkills (param) {
      return this.$lodash.orderBy(this.skills, ['wantsTo', 'name'], ['desc', 'asc']) // TODO Figure out how lodash should be handled in tests
    }
  },
  components: {
    SkillRow,
    ProjectRow,
    UserProfileCard
  },
  data () {
    return {
      basicInfo: {
        name: 'Nicholas Cage',
        title: 'Actor',
        mail: 'jari.sillanpaa@codento.com',
        phone: '050 555 5555',
        aboutMe: 'Bacon ipsum dolor amet salami ham hock rump pork capicola short ribs ground round cupim landjaeger sirloin buffalo t-bone. Sirloin alcatra ham frankfurter t-bone pig spare ribs. Buffalo flank salami rump bacon short ribs meatball sirloin jowl alcatra. Tenderloin kevin venison landjaeger, tongue kielbasa ribeye prosciutto chicken salami pork chop doner ham hock pork belly pig.',
        photo: 'https://baconmockup.com/300/200',
        accounts: [
          {type: 'facebook', address: 'linkedin.com/testprofile'},
          {type: 'linkedIn', address: 'linkedin.com/testprofile'},
          {type: 'twitter', address: 'linkedin.com/testprofile'}]
      },
      skills: [
        {
          name: 'Vue.js',
          knows: 0,
          wantsTo: 2,
          id: 7
        },
        {
          name: 'Node.js',
          knows: 2,
          wantsTo: 1,
          id: 8
        },
        {
          name: 'JavaScript',
          knows: 2,
          wantsTo: 1,
          id: 0
        },
        {
          name: 'React.js',
          knows: 5,
          wantsTo: 2,
          id: 1
        },
        {
          name: 'CoffeeScript',
          knows: 4,
          wantsTo: 1,
          id: 2
        },
        {
          name: 'Angular v.4',
          knows: 3,
          wantsTo: 2,
          id: 3
        },
        {
          name: 'TypeScript',
          knows: 3,
          wantsTo: 1,
          id: 4
        },
        {
          name: 'Java',
          knows: 4,
          wantsTo: 0,
          id: 5
        },
        {
          name: 'Python',
          knows: 3,
          wantsTo: 2,
          id: 6
        }
      ],
      projects: [
        {
          projectName: 'Teuvon telakoneet',
          title: 'Arkkitehti',
          projectID: '123',
          projectDescription: 'Spicy jalapeno venison landjaeger doner short loin, boudin pork loin swine. Strip steak jerky fatback tail. Fatback brisket landjaeger, kielbasa swine capicola venison ham boudin short loin cow sirloin. T-bone prosciutto leberkas sausage capicola kevin porchetta, brisket beef ribs pork belly meatloaf shankle drumstick. Beef venison brisket fatback prosciutto tail jerky shankle boudin pancetta tenderloin biltong. Spare ribs brisket shank jowl jerky. Biltong pig pork belly pork alcatra venison.',
          projectStartDate: '19/10/2015',
          projectEndDate: '20/10/2017'
        },
        {
          projectName: 'Jake järkkääävä',
          projectID: '123',
          title: 'Siivooja',
          projectDescription: 'It’s about making placeholder text great again. That’s what people want, they want placeholder text to be great again. That other text? Sadly, it’s no longer a 10. This placeholder text is gonna be HUGE. Look at these words. Are they small words? And he referred to my words - if they\'re small, something else must be small. I guarantee you there\'s no problem, I guarantee.',
          projectStartDate: '29/10/2015',
          projectEndDate: '20/11/2017'
        },
        {
          projectName: 'Ramin riemut',
          projectID: '123',
          title: 'Iso Kiho',
          projectDescription: 'Well, the way they make shows is, they make one show. That show\'s called a pilot. Then they show that show to the people who make shows, and on the strength of that one show they decide if they\'re going to make more shows. Some pilots get picked and become television programs. Some don\'t, become nothing. She starred in one of the ones that became nothing.',
          projectStartDate: '19/9/2016',
          projectEndDate: '20/10/2016'
        },
        {
          projectName: 'Heikki Hipsterin Kahvila',
          projectID: '123',
          title: 'Barista',
          projectDescription: 'Lyft mixtape air plant raw denim, iceland single-origin coffee semiotics jean shorts man braid live-edge bespoke forage. Trust fund whatever cold-pressed cred, chia hammock butcher single-origin coffee actually truffaut ramps edison bulb neutra vice XOXO. Fanny pack seitan shaman thundercats gastropub butcher hexagon narwhal microdosing fam PBR&B squid. Woke brooklyn literally, ennui paleo street art mlkshk truffaut VHS celiac wolf. Taiyaki neutra palo santo stumptown chillwave meggings authentic brunch mixtape. Yr lomo ugh actually, viral copper mug food truck pour-over lumbersexual seitan helvetica knausgaard hammock austin. Tacos franzen slow-carb blue bottle tumblr literally organic godard next level.',
          projectStartDate: '19/9/2015',
          projectEndDate: '20/9/2015'
        }
      ]
    }
  },

  methods: {
    getKnowledge (knows) {
      switch (knows) {
        case 1:
          return { text: 'Alkeet', class: 'text-warning' }
        case 2:
          return { text: 'Duunari', class: 'text-primary' }
        case 3:
          return { text: 'Seniori', class: 'text-primary' }
        case 4:
          return { text: 'Arkkitehti', class: 'text-success' }
        case 5:
          return { text: 'Guru', class: 'text-success' }
        default:
          return { text: 'Ei osaa', class: 'text-warning' }
      }
    },
    getWants (wantsTo) {
      switch (wantsTo) {
        case 0:
          return { text: 'En halua tehdä', class: 'text-danger' }
        case 1:
          return { text: 'Voin tehdä varauksella', class: 'text-primary' }
        case 2:
          return { text: 'Haluan tehdä', class: 'text-success' }
        default:
          return { text: 'Arvoa ei ole annettu', class: 'text-warning' }
      }
    }
  }
}
</script>

<!-- Add 'scoped' attribute to limit CSS to this component only -->
<style scoped>

</style>
