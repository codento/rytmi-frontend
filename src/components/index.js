import Aside from './appStructures/Aside.vue'
import Breadcrumb from './appStructures/Breadcrumb.vue'
import Callout from './helpers/Callout.vue'
import Footer from './appStructures/Footer.vue'
import Header from './appStructures/Header.vue'
import Sidebar from './appStructures/Sidebar/Sidebar.vue'

export {
  Aside,
  Breadcrumb,
  Callout,
  Footer,
  Header,
  Sidebar
}
