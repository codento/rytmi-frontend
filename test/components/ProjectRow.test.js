import { mount } from 'vue-test-utils'
import { ProjectRow } from '../../src/components/Profile'

describe('UserProfileCard.test.js', () => {
  it('Template is correct', () => {
    let wrapper = mount(ProjectRow, {
      propsData: {
        project: {
          projectName: 'Testi prkkis',
          title: 'mail',
          projectStartDate: '15.05.2017',
          projectEndDate: '16.06.2017',
          projectprojectDescription: 'Hipster ipsum'
        }
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
