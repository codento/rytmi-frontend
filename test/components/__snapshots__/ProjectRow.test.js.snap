// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileCard.test.js Template is correct 1`] = `
"<div class=\\"row mb-1\\"><div class=\\"col-sm-12\\"><span style=\\"font-size: 24px;\\">mail </span></div> <div class=\\"col-sm-12\\"><span style=\\"color: rgb(248, 149, 0); font-size: 16px;\\">
        Testi prkkis  15.05.2017 - 16.06.2017 
      </span></div> <div class=\\"col-sm-12\\">
    w
  </div></div>"
`;
